xaynetwork/xayn_discovery_app
# Tb 3998 add duration param to bottom sheet dismissed analytics event

### What 🕵️ 🔍

- Track screen duration for bottom sheets, and send duration (in seconds) with `bottomSheetDismissed` event.
- Move Overlay related files to `overlay` directory in utils.

I didn't rely on overlay manager/mixin to send the `bottomSheetDismissed` event, even though it could be a cleaner way, because 
1. we have more than a bottom sheet that could be triggered with a single `OverlayManager` instance in cases of stacking multiple bottom sheets on top of each other.
2. the "OverlayData" is not fully exposed to the manager to be read.
3. We only want to track onboarding and collection bottom sheet dismiss (cancel) events, not all the app's bottom sheets.

----------

### How to test 🥼 🔬
  - [x] Check OnBoarding bottom sheets are working as expected.
  - [ ] Check Collection bottom sheets are working as expected.
 
  Check that `bottomSheetDismissed` events on Mixpanel have `duration` param when you click on "cancel" for the following bottom sheets:
   - [ ] saveToCollection
  - [ ] moveMultipleBookmarksToCollection
  - [ ] createCollection
  - [ ] renameCollection
  - [ ]   confirmDeletingCollection
  - [ ]  onBoardingCollectionOptions
   - [ ]  onBoardingManageCollection
   - [ ] onBoardingManageBookmark
  - [ ]  onBoardingLikeAndDislike
  - [ ]  onBoardingNextArticle
  - [ ]  onBoardingLongPressSaveArticle

----------

### References 📝 🔗

- [JIRA](https://xainag.atlassian.net/browse/TB-3998)

----------
